modules/AlphaWalletCore/AlphaWalletCore/WebImageURL.swift

```diff
@@ -84,17 +84,34 @@ public enum WebImageURL: Codable, Hashable, Equatable, CustomStringConvertible {
     }
 
     public init(url: URL, withUrlRewriting: Bool = true, rewriteGoogleContentSizeUrl size: GoogleContentSize = .s750) {
-        if let url = WebImageURL.functional.rewriteGoogleContentSizeUrl(url: url, size: size), withUrlRewriting {
+        let maybeIpfs = url.absoluteString
+        var rewrittenUrl: URL
+        if maybeIpfs.trimmed.nonEmpty && maybeIpfs.trimmed.range(of: "^[a-zA-Z0-9]*$", options: .regularExpression) != nil {
```

Comment:
Do you think we can introduce a simple heuristic to reject completely invalid ones by doing this:

 instead of "*" or "+" make it match with at least 16 characters? Since it's a base encoded (58 etc. https://docs.ipfs.tech/concepts/hashing/#important-hash-characteristics) of a hash, seems like it would never be lower.

If we do, let's add a comment to that magic number "16".